Examples/WindowsFormsApplication/Nfield.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Win32;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WindowsFormsApplication
+{
+    public class Nfield
+    {
+        private IPublicClientApplication ClientApp { get; init; }
+        private string[] Scopes { get; } = new[] { CreateScope("user_impersonation"), CreateScope("Survey.Read") };
+        public string DomainName { get; }
+
+        private INfieldConnectionV2 NfieldConnection { get; set; }
+
+        public Nfield(string domainName)
+        {
+            ClientApp = CreateClientApp();
+            DomainName = domainName;
+
+            NfieldConnection = NfieldConnectionFactory.Create(ApplicationConfiguration.Current.NfieldApiUri);
+            NfieldConnection.RegisterTokenProvider(domainName, ProvideTokenAsync);
+        }
+
+        private IPublicClientApplication CreateClientApp()
+        {
+            var client = PublicClientApplicationBuilder.Create(ApplicationConfiguration.Current.ClientId)
+                .WithRedirectUri("http://localhost")
+                .WithAuthority(AzureCloudInstance.AzurePublic, ApplicationConfiguration.Current.Tenant)
+                .Build();
+
+            client.UserTokenCache.SetBeforeAccess(BeforeAccessNotification);
+            client.UserTokenCache.SetAfterAccess(AfterAccessNotification);
+
+            return client;
+        }
+
+        private async Task<string> ProvideTokenAsync()
+        {
+            var result = await AuthenticateAsync((status) => { });
```

Comment:
Maybe some extra context here on why an empty anonymous function is sufficient.

Reply:
Added comment in code and extracted the noop to a local variable